### DISCUSSION TITLE
simplify defining nested types

### DISCUSSION THREAD

**Discussion by ghost** [dotnet/csharplang#7958] [General]
I would like to define ```Container``` in file ```Container.cs``` and ```Container.Nested``` in file ```Container.Nested.cs```.

Now, it's like this.
##### Container.cs
```
    public partial class Container
    {

    }
```
##### Container.Nested.cs
```
    public partial class Container
    {
        public class Nested
        {

        }
    }
```

Could this be simplified like this.
##### Container.cs
```
    public class Container
    {

    }
```
##### Container.Nested.cs
```
    public class Container.Nested
    {

    }
```

**Comment by HaloFour** [comment]
See: #711
